game/src/info/trip.rs

```diff
@@ -386,6 +389,329 @@ pub fn cancelled(ctx: &mut EventCtx, app: &App, id: TripID) -> Widget {
     Widget::col(col)
 }
 
+/// Highlights intersections which were "slow" on the map
+fn highlight_slow_intersections(app: &App, details: &mut Details, id: TripID, ctx: &EventCtx) {
+    let data_opt = &app.primary.sim.get_analytics().trip_intersection_delays(id);
+    if data_opt.is_some() {
+        let data = data_opt.unwrap();
+        for (id, time) in data {
+            let intersection = app.primary.map.get_i(id.parent);
+            // Maybe alter the delay times
+            let (normal_delay_time, slow_delay_time) = if intersection.is_traffic_signal() {
+                (30, 120)
+            } else {
+                (5, 30)
+            };
+            let (fg_color, bg_color) = if time < &normal_delay_time {
+                (Color::WHITE, app.cs.normal_slow_intersection)
+            } else if time < &slow_delay_time {
+                (Color::BLACK, app.cs.slow_intersection)
+            } else {
+                (Color::WHITE, app.cs.very_slow_intersection)
+            };
+            // Draw label after, to prevent being obscured by the line
+            let time_duration = Duration::seconds(*time as f64);
+            let unzoomed_label = Text::from(TextSpan::from(
+                Line(format!("{}", time_duration)).body().fg(fg_color),
+            ))
+            .bg(bg_color);
+            let zoomed_label = Text::from(TextSpan::from(
+                Line(format!("{}", time_duration)).size(6).fg(fg_color),
+            ))
+            .bg(bg_color);
+            let unzoomed_rendered = unzoomed_label
+                .render(ctx)
+                .centered_on(intersection.polygon.center());
+            let zoomed_rendered = zoomed_label
+                .render(ctx)
+                .centered_on(intersection.polygon.center());
+            details.unzoomed.append(unzoomed_rendered.clone());
+            details.zoomed.append(zoomed_rendered);
+        }
+    }
+}
+
+/// Highlights lanes which were "slow" on the map
+fn highlight_slow_lanes(app: &App, details: &mut Details, id: TripID, ctx: &EventCtx) {
+    let data_opt = &app.primary.sim.get_analytics().trip_lane_speeds(id);
+    if data_opt.is_some() {
+        let data = data_opt.unwrap();
+        for (id, speed_percent) in data {
+            let lane = app.primary.map.get_l(*id);
+            let (fg_color, bg_color) = if speed_percent > &95 {
+                (Color::WHITE, app.cs.normal_slow_intersection)
+            } else if speed_percent > &60 {
+                (Color::BLACK, app.cs.slow_intersection)
+            } else {
+                (Color::WHITE, app.cs.very_slow_intersection)
+            };
+            details.unzoomed.push(
+                Fill::Color(bg_color),
```

Comment:
You can directly pass `bg_color` here. The signature of this call is `pub fn push<F: Into<Fill>>(&mut self, fill: F, p: Polygon) {`, which is a Rust idiom you might've seen before in methods that either take `&str` or `String`. `Color` implements `Into<Fill>`, so the implicit conversion works here too.

Reply:
Oh nice!

Reply:
That's pretty handy, changed.